packages/core/src/tools/grep.ts

```diff
@@ -23,7 +25,114 @@ import { getErrorMessage, isNodeError } from '../utils/errors.js';
 import { isGitRepository } from '../utils/gitUtils.js';
 import { Config } from '../config/config.js';
 
-// --- Interfaces ---
+/**
+ * Memory safety constants to prevent heap overflow
+ */
+const MEMORY_SAFETY = {
+  /** Maximum file size to process (50MB - conservative limit for memory safety) */
+  MAX_FILE_SIZE: 50 * 1024 * 1024,
+  /** Maximum number of files to process in JavaScript fallback */
+  MAX_FILES_TO_PROCESS: 10000,
```

Comment:
worried about this limit. this will break grep being useful for very large projects.  can we relax this somewhat